cirq-google/cirq_google/ops/analog_detune_gates.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2025 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define detuning gates for Analog Experiment usage."""
+from __future__ import annotations
+
+from typing import AbstractSet, Any, TYPE_CHECKING, TypeAlias
+
+import sympy
+import tunits as tu
+
+import cirq
+
+if TYPE_CHECKING:
+    import numpy as np
+
+# The gate is intended for the google internal use, hence the typing style
+# follows more on the t-unit + symbol instead of float + symbol style.
+ValueOrSymbol: TypeAlias = tu.Value | sympy.Basic
+FloatOrSymbol: TypeAlias = float | sympy.Basic
+
+
+@cirq.value_equality(approximate=True)
+class AnalogDetuneQubit(cirq.ops.Gate):
+    """A step function that steup a qubit to the freq according to analog model.
+    Pulse shape:
+    .. svgbob::
+      :align: center
+                   |   ,--------|---- amp (calculated from target freq using analog model)
+                   |  /         |
+        prev_amp---|-' - - - - -| - -
+        ^
+        |          |-w -|       |
+        |          |---length --|
+        |
+        --------------------------(calculated from previous qubit freq using analog model)
+    Because the step hold at amp with inifinite length. You need prev_freq to compensate
```

Comment:
This sentence needs to be adjusted "Because the step hold at amp with inifinite length.".  I am not quite sure what you mean.  Is it "Because the step is held at amp with infinite length, you need prev_freq ..."
(Also infinite is a typo)

Reply:
Sorry for the confusing. Behind it is DetuneStepUp logic, that DAC amp is always on). 

Update it into this:
> Note that the step is held at a constant amplitude indefinitely. This gate is typically used by concatenating multiple instances. To ensure the curve is continuous and avoids sudden jumps, you need prev_freq to compensate for the previous Detune gate.